chore: [IA-21] added permissions mapping in README (#3132)
pagopa/io-app: README.md

If you have some time to spare and wish to get involved on a regular basis, [contact us](mailto:federico.feroldi@pagopa.it).

### What permissions are used by the IO app?

| Permission                        | Usage                                                                   |
| --------------------------------- | ---------------------------------------------------------------------   |
| Calendar                          | Request access to a user’s calendar data to save reminders              |
| Camera                            | Request access to the device camera, used to scan codes and documents   |
| Face ID                           | Request permission to use biometric authentication                      |

> | Fingerprint                       | Request permission to use fingerprint sensor for authentication         |

Author: I think so. It's a better description and is a more appropriate term.

> Identification is the ability to identify uniquely a user of a system or an application that is running in the system. Authentication is the ability to prove that a user or application is genuinely who that person or what that application claims to be. (cit. [IBM MQ](https://www.ibm.com/docs/en/ibm-mq/9.1?topic=mechanisms-identification-authentication))

If you have some time to spare and wish to get involved on a regular basis, [contact us](mailto:federico.feroldi@pagopa.it).

### What permissions are used by the IO app?

> Because different platform have different types of Permissions below we have two sections about permissions requested by the IO app for both environment (iOS and Android). Some permissions may be defined but not used. Their presence is due to dependencies with third-party modules or because they are required by the target store.

Reviewer: platform -> platforms
environment -> environments

Yes, it's me <sub><sup>nitpicking grammar on Github</sup></sub> (⌐■_■)

Author: ```suggestion
Because different platforms have different types of Permissions below we have two sections about permissions requested by the IO app for both environments (iOS and Android). Some permissions may be defined but not used. Their presence is due to dependencies with third-party modules or because they are required by the target store.
```

**android** 

| Permission (android.permission.*)| Usage / Meaning                                                                                                                |
| -------------------------------- | --------------------------------------------------------------------------------------------------------------------- |
| INTERNET                         | Allows applications to open network sockets (e.g. simple internet connectivity)                                       |
| ACCESS_NETWORK_STATE             | Allows applications to access information about networks (e.g. details about connection quality/state)                |
| ACCESS_WIFI_STATE                | Allows applications to access information about WIFI state                                                            |
| CAMERA                           | Allows applications to access device camera to scan QR codes                                                          |
| FOREGROUND_SERVICE               | Allows applications to use foreground service                                                                         |

> | MODIFY_AUDIO_SETTINGS            | Allows an application to modify global audio settings                                                                 |

Reviewer 2: ```suggestion
| MODIFY_AUDIO_SETTINGS            | Allows an application to modify global audio settings. Related to camera usage                                                                 |
```

| NFC                              | Allows applications to perform I/O operations over NFC                                                                |

> | RECEIVE_BOOT_COMPLETED           | Allows an application to receive the Intent.ACTION_BOOT_COMPLETED that is broadcast after the system finishes booting |

Reviewer 2: ```suggestion
| RECEIVE_BOOT_COMPLETED           | Allows an application to receive the Intent.ACTION_BOOT_COMPLETED that is broadcast after the system finishes booting. Used for push notification |
```

| VIBRATE                          | Allows access to the vibrator. This allow the application to emit vibration                                           |

> | WAKE_LOCK                        | Allows using PowerManager WakeLocks to keep processor from sleeping or screen from dimming                            |

Reviewer 2: ```suggestion
| WAKE_LOCK                        | Allows using PowerManager WakeLocks to keep processor from sleeping or screen from dimming. Used for push notification                            |
```

| READ_APP_BADGE                   | Notification Badges that show on app icons                                                                            |
| READ_CALENDAR                    | Allows an application to read the user's calendar data                                                                |

> | WRITE_CALENDAR                   | Allows an application to write the user's calendar data                                                               |

Reviewer 2: ```suggestion
| WRITE_CALENDAR                   | Allows an application to write the user's calendar data. Used to automatically set reminders.                                                               |
```

> | READ_EXTERNAL_STORAGE            | Allows an application to read from external storage                                                                   |

Reviewer 2: ```suggestion
| READ_EXTERNAL_STORAGE            | Allows an application to read from external storage. Used to pick images from gallery with payment QRCode                                                                   |
```

> | WRITE_EXTERNAL_STORAGE           | Allows an application to write to external storage                                                                    |

Reviewer 2: ```suggestion
| WRITE_EXTERNAL_STORAGE           | Allows an application to write to external storage. Used to store images, e.g.: save bonus information (QRCode for Bonus Vacanze or EuCovid Certificate)                                                                 |
```

> | USE_FINGERPRINT                  | Allows an app to use fingerprint hardware for a faster identification                                                 |

Reviewer 2: ```suggestion
| USE_FINGERPRINT                  | Allows an app to use fingerprint hardware for biometric identification                                                 |
```

Reviewer 2: I don't if it is faster or not 😄

Below there are the permissions required by the main android hardware manufacturers. Mainly used to manage notification badge icons.

| Permission (manufacturer)                                               | Usage / Meaning                                                                              |
| ----------------------------------------------------------------------- | --------------------------------------------------------------------------------- |
| com.anddoes.launcher.permission.UPDATE_COUNT                            | used for the notification badge                                                   |

> | com.google.android.c2dm.permission.RECEIVE                              | It is used when receiving a broadcast from GCM server that contains a GCM message |

Reviewer 2: ```suggestion
| com.google.android.c2dm.permission.RECEIVE                              | It is used when receiving a broadcast from GCM server that contains a GCM message. Used for push notification |
```

| com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE  | It is used by Firebase to recognize where the app was installed from              |

> | com.htc.launcher.permission.READ_SETTINGS                               | used for the notification badge                                                   |

Reviewer 2: ```suggestion
| com.htc.launcher.permission.READ_SETTINGS                               | used for the notification badge. Specific for HTC vendor                                                  |
```

| com.htc.launcher.permission.UPDATE_SHORTCUT                             | used for the notification badge                                                   |

> | com.huawei.android.launcher.permission.CHANGE_BADGE                     | used for the notification badge                                                   |

Reviewer 2: ```suggestion
| com.huawei.android.launcher.permission.CHANGE_BADGE                     | used for the notification badge. Specific for Huawei vendor                                                   |
```

| NSBluetoothAlwaysUsageDescription           | This key is required if your app uses the device’s Bluetooth interface.                                                                  |
| NSBluetoothPeripheralUsageDescription       | This key is required if your app uses APIs that access Bluetooth peripherals and has a deployment target earlier than iOS 13.            |
| NSContactsUsageDescription                  | IO needs access to your contacts to let you add them in calendar events.                                                                 |
| NSLocationAlwaysUsageDescription            | This key is required if your iOS app uses APIs that access the user’s location at all times and deploys to targets earlier than iOS 11.  |
| NSLocationUsageDescription                  | This key is required if your app uses APIs that access the user’s location information.                                                  |
| NSLocationWhenInUseUsageDescription         | This key is required if your iOS app uses APIs that access the user’s location information while the app is in use.                      |

> | NSMicrophoneUsageDescription                | IO needs access to the microphone in case you want to leave a voice note.                                                                |

Reviewer 2: ```suggestion
| NSMicrophoneUsageDescription                | IO needs access to the microphone in case you want to leave a voice note. Used in the assistance flow                                                               |
```

| NSMotionUsageDescription                    | This key is required if your app uses APIs that access the device’s motion data.                                                         |
| NSCalendarsUsageDescription                 | IO needs access to the calendar to add event reminders.                                                                                  |
| NSCameraUsageDescription                    | IO needs access to the camera to scan QR codes.                                                                                          |

> | NSFaceIDUsageDescription                    | Enable Face ID for a faster identification.                                                                                              |

Reviewer 2: ```suggestion
| NSFaceIDUsageDescription                    | Enable Face ID for biometric identification.                                                                                              |
```

| NSPhotoLibraryAddUsageDescription           | This key is required if your app uses APIs that have write access to the user’s photo library.                                           |
| NSPhotoLibraryUsageDescription              | IO needs access to the Photos to scan QR codes.                                                                                          |

> | NSSpeechRecognitionUsageDescription         | IO needs access to the Photos to scan QR codes.                                                                                          |

Reviewer: [NSSpeechRecognitionUsageDescription](https://developer.apple.com/documentation/bundleresources/information_property_list/nsspeechrecognitionusagedescription) should have a different usage description:

> This key is required if your app uses APIs that send user data to Apple’s speech recognition servers.

Author: ```suggestion
| NSSpeechRecognitionUsageDescription         | This key is required if your app uses APIs that send user data to Apple’s speech recognition servers. Used in the assistance flow.                                                                                        |
```
